### DISCUSSION TITLE
Airflow as Solution

### DISCUSSION THREAD

**Discussion by Sakki86** [apache/airflow#26184] [Q&A]
**### Hello Everyone,

Have couple of questions on Airflow jobs and how this tool can be a good solution for some scheduling options we are looking for.

Currently we have an Oracle job, that executes multiple PL SQL jobs and runs aggregations on granular level data , along with transformations on some objects and it is quite time extensive process. We are planning to move to event based architecture that calls multiple dataflow jobs as part of scheduling or triggering that event. I heard DAG is the collection of all tasks associated with an event and executes the dependent tasks as Cronjob... 

Wondering if this is a good tool to help our needs. Anyone please suggest PROS and CONS of Airflow in terms of Monitoring, Dependencies/Scalability etc.**

**Comment by potiuk** [comment]
Yes. Airflow is all of that.

To be honest, this is so broad question that there is no way to answer it, but to send you to the docs, asking you to follow it, do the quick start and tutorial and asses yourself if it is a good fit. Start here: https://airflow.apache.org/docs/apache-airflow/stable/index.html

All you wrote above is true. And it seems to fit well granularity you describe. And Airlfow is very scalable, can plugin into monitoring and have rich set of dependency management. Those are PROS. You can read more detail about Airflow's concepts here https://airflow.apache.org/docs/apache-airflow/stable/concepts/index.html 

It's a software you need to manage - this is CON, but you can mitigate it by choosing the right installation options  https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html by reading the docs you can choose the right option to mitigate any complexity in deployment you want to avoid.
